extensions/test-src/org/pentaho/platform/web/http/api/resources/UserSettingsResourceTest.java

```diff
@@ -46,8 +56,29 @@ public void teardown() {
   public void testSetUserSetting() throws Exception {
     IUserSettingService userSettingService = mock( IUserSettingService.class );
     PentahoSystem.registerObject( userSettingService );
-    Response response = userSettingsResource.setUserSetting( "favorites", "[{\"fullPath\":\"/public/Steel Wheels/Top Customers (report).prpt\", \"title\":\"<script>if(true&&'str'!='str2')alert(1)</script>\", \"lastUse\":0}]" );
+    Response response = userSettingsResource.setUserSetting( USER_SETTING_NAME, "[{\"fullPath\":\"/public/Steel Wheels/Top Customers (report).prpt\", \"title\":\"<script>if" + "(true&&'str'!='str2')alert(1)</script>\", \"lastUse\":0}]" );
+
+    verify( userSettingService ).setUserSetting( USER_SETTING_NAME,      USER_SETTING_VALUE );
+  }
+
+  @Test
+  public void testGetUserSettings() throws Exception {
+    IUserSettingService userSettingService = mock( IUserSettingService.class );
+
+    IUserSetting userSetting = mock( IUserSetting.class );
+    when( userSetting.getSettingName() ).thenReturn( USER_SETTING_NAME );
+    when( userSetting.getSettingValue() ).thenReturn( USER_SETTING_VALUE );
+
+    List<IUserSetting> userSettings = new ArrayList<>();
+    userSettings.add( userSetting );
+    when( userSettingService.getUserSettings() ).thenReturn( userSettings );
+
+    PentahoSystem.registerObject( userSettingService );
 
-    verify( userSettingService ).setUserSetting( "favorites", "[{\"fullPath\":\"/public/Steel Wheels/Top Customers (report).prpt\",\"title\":\"\\u003Cscript\\u003Eif(true\\u0026\\u0026\\u0027str\\u0027!=\\u0027str2\\u0027)alert(1)\\u003C/script\\u003E\",\"lastUse\":0}]" );
+    ArrayList<Setting> response = userSettingsResource.getUserSettings();
```

Comment:
This test does not actually test anything. You're mocking this method and invoking it here. It does not touch the actual class.